### PR TITLE
Initial commit with working version.

### PR DESCRIPTION
**Pull request by marianamarasoiu**

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+name: 'freckl'
+version: 0.0.1
+description: An absolute bare-bones web app.
```
Comment (lukechurch) [inline pubspec.yaml]: It isn't.

File: pubspec.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+#author: <your name> <email@example.com>
+#homepage: https://www.example.com
+environment:
```
Comment (lukechurch) [inline pubspec.yaml:5]: I think you should change your name

File: web/index.html [hunk 3]
```diff
@@ -0,0 +1,40 @@
+<!DOCTYPE html>
+
+<!--
+  Copyright (c) 2015, <your name>. All rights reserved. Use of this source code
```
Comment (lukechurch) [inline web/index.html]: Your name isn't getting any better.

File: web/main.dart [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  panZoom
+    ..zoomEnabled = true
+    ..panEnabled = true
+    ..zoomSensitivity = 0.02;
+}
+
+void buttonPressHandler(MouseEvent event) {
+  InputElement input = querySelector('#path-to-json');
+  String pathToJson = input.value;
+  HttpRequest.getString(pathToJson).then((String response) {
```
Comment (lukechurch) [inline web/main.dart]: Can you use async/await here?

I.e. Mark the function as async, and 

String response = await HttpRequest.getString(pathToJson);

File: web/main.dart [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+  svg.SvgSvgElement innerSvg = querySelector('.inner');
+  svg.GElement group = innerSvg.querySelector('#viewport');
+  group.nodes.clear();
+
+  // Record maximum values of x and y which represent the width and height for
+  // background necessary for zooming and panning.
+  num maxX, maxY;
+  maxX = maxY = double.NEGATIVE_INFINITY;
+
+  for (Map dataPoint in data) {
```
Comment (lukechurch) [inline web/main.dart:37]: Could you add a sanity check on the format of dataPoint here, so that we get debug data that makes sense fi there's a failure?

**Comment by lukechurch** [comment]
lgtm with some suggestions - this looks great!

File: web/main.dart (hunk 4 above)
Comment (lukechurch) [inline web/main.dart]: Sorry, ignore this, listen wants void, which means that you'd have to do contortions to avoid returning a future.

**Comment by marianamarasoiu** [comment]
Done. PTAL and merge if it looks good :)

**Pull request merged**